app/code/Magento/Catalog/Model/Category.php

```diff
@@ -796,11 +796,17 @@ public function getAllChildren($asArray = false)
      * @param boolean $recursive
      * @param boolean $isActive
      * @param boolean $sortByPosition
-     * @return string
+     * @param boolean $asArray return result as array instead of comma-separated list of IDs
+     * @return array|string
      */
-    public function getChildren($recursive = false, $isActive = true, $sortByPosition = false)
+    public function getChildren($recursive = false, $isActive = true, $sortByPosition = false, $asArray = false)
```

Comment:
We are not allowed to add new parameters to public methods according to [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/#php)

Reply:
This shouldn't create a compatibility issue because it defaults to `false` meaning the default behaviour is still the same.

Reply:
@cruxicheiros it does not matter, guide is clear about that. it is such strict to prevent a situation when additional parameters were added in custom code.

Reply:
@cruxicheiros yes, separate method is the best way to resolve it